CONTRIBUTING.md

```diff
@@ -125,3 +103,31 @@ Automated tests are currently crude and limited in functionality, but improvemen
 Currently, to run tests you need to first have started local test databases (e.g. using `npm run db:start`).
 
 The databases can take a few seconds to start up, so you might need to give it a minute before running the tests.
+
+## For maintainers
+
+We use [semantic-release](https://github.com/semantic-release/semantic-release) together with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0) to automate releases. This makes the maintainenance process easier and less error-prone to human error.
+
+When accepting Pull Requests, make sure the following:
+
+* Use "Squash and merge"
+* Rewrite the commit message to conform to `Conventional Commits` style. Check the "Recommended Scopes" section for further advice.
+* Optionally link issues the PR will resolve (you can add "closed" in front of the issue numbers to close the issues automatically, when the PR is merged)
+
+
+### Recommended Scopes
+
+A typical conventional commit looks like this:
+```
+type(scope): title
+
+body
+```
+
+Scope is the part that will help with groupping the different commit types in the release notes.
+
+- **provider** - Provider related changes. (eg.: "feat(provider): add X provider", "docs(provider): fix typo in X documentation"
+- **database** - Database related changes. (eg.: "feat(database): add X database", "docs(database): fix typo in X documentation"
```

Comment:
Any other recommended scopes?

cc: @nextauthjs/core

Reply:
Hmm, maybe "core" as a scope as well?

Reply:
Thought of it, but I feel it will be overused, and and thus unnecessary, as it won't "group" anything.

Reply:
Yeah was just an idea, kind of like a placeholder for all the other stuff that has to do with next-auth itself, not providers or db's, etc. know what I mean?

Reply:
yes. but since it's "everything else", it would just be noise, I feel. Core changes are also not necessarily interesting for our users directly.

Reply:
Yeah good point. Then lets just leave it at those "secondary" scopes.